Common/Api/OptimizationBacktestJsonConverter.cs

```diff
@@ -0,0 +1,221 @@
+            {
+                { "Alpha", jStatistics[0].Value<string>() },
+                { "Annual Standard Deviation", jStatistics[1].Value<string>() },
+                { "Annual Variance", jStatistics[2].Value<string>() },
+                { "Average Loss", jStatistics[3].Value<string>() },
+                { "Average Win", jStatistics[4].Value<string>() },
+                { "Beta", jStatistics[5].Value<string>() },
+                { "Compounding Annual Return", jStatistics[6].Value<string>() },
+                { "Drawdown", jStatistics[7].Value<string>() },
+                { "Estimated Strategy Capacity", jStatistics[8].Value<string>() },
+                { "Expectancy", jStatistics[9].Value<string>() },
+                { "Information Ratio", jStatistics[10].Value<string>() },
+                { "Loss Rate", jStatistics[11].Value<string>() },
+                { "Net Profit", jStatistics[12].Value<string>() },
+                { "Probabilistic Sharpe Ratio", jStatistics[13].Value<string>() },
+                { "Profit-Loss Ratio", jStatistics[14].Value<string>() },
+                { "Sharpe Ratio", jStatistics[15].Value<string>() },
+                { "Total Fees", jStatistics[16].Value<string>() },
+                { "Total Trades", jStatistics[17].Value<string>() },
+                { "Tracking Error", jStatistics[18].Value<string>() },
+                { "Treynor Ratio", jStatistics[19].Value<string>() },
+                { "Win Rate", jStatistics[20].Value<string>() },
+            };
+
+            var parameterSet = new ParameterSet(-1, serializer.Deserialize<Dictionary<string, string>>(jObject["parameterSet"].CreateReader()));
+
+            var equity = new Series { Values = GetSeriesValues(jObject["equity"]) };
+
+            var optimizationBacktest = new OptimizationBacktest(parameterSet, backtestId, name)
+            {
+                HostName = hostName,
+                Progress = progress,
+                ExitCode = exitCode,
+                Statistics = statistics,
+                Equity = equity
+            };
+
+            return optimizationBacktest;
+        }
+
+        /// <summary>
+        /// Get x and y value pairs that represent series data
+        /// </summary>
+        /// <param name="values">json array of x, y value pairs</param>
+        /// <returns>List of ChartPoints</returns>
+        private static List<ChartPoint> GetSeriesValues(JToken values)
+        {
+            var chartPoints = new List<ChartPoint>();
+
+            foreach (var point in values.Children())
+            {
+                var x = point[0];
+                var y = point[1];
+
+                if (y != null && y.Type == JTokenType.Float)
+                {
+                    chartPoints.Add(new ChartPoint((long)x, (decimal)y));
+                }
+                else
+                {
+                    chartPoints.Add(new ChartPoint((long)x, (int)y));
```

Comment:
Can also really convert int into the decimal above? avoiding the branching

Reply:
I did a few refactors in that code but missed this, thanks!